nemo/collections/asr/parts/utils/transcribe_utils.py

```diff
@@ -421,3 +438,21 @@ def transcribe_partial_audio(
             asr_model.decoder.unfreeze()
         logging.set_verbosity(logging_level)
     return hypotheses
+
+
+def separate_punctuation(line):
+    punctuation_marks = '.,?'
+    regex_separate_punctuation = fr"([{''.join(punctuation_marks)}])"
+    line = re.sub(regex_separate_punctuation, r' \1 ', line)
+    return line
+
+
+def preprocess(line, do_lowercase, rm_punctuation):
+    line = line.replace("\n", " ").strip()
+    if do_lowercase:
+        line = line.lower()
+    if rm_punctuation:
```

Comment:
Don't manually replace P&C. Use above list and loop through it to replace all P&C with space or other character

Reply:
Done